.changeset/witty-papers-tan.md

```diff
@@ -0,0 +1,124 @@
+
+## Key Benefits
+
+- **Synchronous Creation**: Allows for the immediate creation of child datastores without waiting for asynchronous operations.
+- **Strong Typing**: Ensures type safety and better developer experience by leveraging TypeScript's type system.
+
+## Use Cases
+
+### Example 1: Creating a Child Datastore
+
+In this example, we demonstrate how to support creating a child datastore synchronously from a parent datastore.
+
+```typescript
+import {
+    IFluidDataStoreFactory,
+    IFluidDataStoreContext,
+    IFluidDataStoreRuntime,
+    IFluidDataStoreChannel,
+} from "@fluidframework/runtime-definitions";
+import { SharedMap } from "@fluidframework/map";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { assert } from "@fluidframework/core-utils";
+
+class ChildDataStore {
+    public static create(runtime: IFluidDataStoreRuntime) {
+        const root = SharedMap.create(runtime, "root");
+        root.bindToContext();
+        return new ChildDataStore(runtime, root);
+    }
+
+    public static async load(runtime: IFluidDataStoreRuntime) {
+        const root = (await runtime.getChannel("root")) as unknown as ISharedMap;
+        return new ChildDataStore(runtime, root);
+    }
+
+    private constructor(
+        private readonly runtime: IFluidDataStoreRuntime,
+        public readonly sharedMap: SharedMap,
+    ) {}
+
+    get handle() {
+        return this.runtime.entryPoint;
+    }
+}
+
+class ChildDataStoreFactory implements IFluidDataStoreFactory {
+    static readonly instance = new ChildDataStoreFactory();
+    private constructor() {}
+
+    get IFluidDataStoreFactory() {
+        return this;
+    }
+
+    public readonly type = "ChildDataStore";
+
+    async instantiateDataStore(context: IFluidDataStoreContext, existing: boolean) {
+        const runtime: IFluidDataStoreRuntime = new FluidDataStoreRuntime(
+            context,
+            sharedObjectRegistry,
+            existing,
+            async () => dataStore,
+        );
+        const dataStore = existing ? ChildDataStore.load(runtime) : ChildDataStore.create(runtime);
+
+        return runtime;
+    }
+
+    createDataStore(context: IFluidDataStoreContext) {
+        runtime: IFluidDataStoreChannel;
+        entrypoint: ChildDataStore;
+    } {
+        const runtime = new FluidDataStoreRuntime(context, new Map(), false, async () => entrypoint);
+        const entrypoint = ChildDataStore.create(runtime);
+        return { runtime, entrypoint };
+    }
+}
+
+class ParentDataObject extends DataObject {
+    protected async initializingFirstTime() {
+        this.createChild("initialChild");
+    }
+
+    createChild(name: string): ChildDataStore {
+        assert(this.context.createChildDataStoreSync !== undefined, "createChildDataStoreSync is not defined");
```

Comment:
it might be worth having some comment here helping developers understand why they might be hitting this assert.
My expectation is that only Loop would care (N/N-1 collab boundary), but you never know